app/controllers/application_controller/wait_for_task.rb

```diff
@@ -11,65 +11,64 @@ def wait_for_task
     @edit = session[:edit]  # If in edit, need to preserve @edit object
     raise Forbidden, _('Invalid input for "wait_for_task".') unless params[:task_id]
 
-    @edit = session[:edit]  # If in edit, need to preserve @edit object
-    session[:async] ||= {}
-    session[:async][:interval] ||= 1000 # Default interval to 1 second
-    session[:async][:params] ||= {}
+    async_interval = params[:async_interval] || 1000 # Default interval to 1 second
+    async_params = params[:async_params] || session[:async][:params]
```

Comment:
I don't think I like the async_params coming in as a regular param - not sure why that is needed.